add .cmap and .norm property to plot_settings (#3624)
sunpy/sunpy: docs/guide/data_types/maps.rst

    plt.colorbar()
    plt.show()

> Earlier, you had and option to view or make changes to the default settings through

Reviewer: Remove this. We don't need to mention deprecated options in the docs

    import sunpy.data.sample
    import matplotlib.pyplot as plt
    smap = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)

>     plot_settings = {}

Reviewer 2: This example doesn't seem to be right? What are we recommending people do now? Pass these in as kwargs to `plot`?

Author: since there is no `plot_settings` property associated with `sunpy.map.Map`, we have to pass the plot-specific attributes like `title` as kwargs to `plot`.
I just realized this can have a better implementation as below:
```python
plot_settings = {'title':'My Second SunPy Plot'}
smap.cmap = plt.cm.Blues_r
fig = plt.figure()
smap.plot()
```

Reviewer 2: My point was you aren't actually passing the `plot_settings` anywhere after you create it. What I think we should be documenting is:
```python
fig = plt.figure()
smap.plot(cmap="Blues_r", title="My Second Plot")
```
and not constructing `plot_settings` or setting `map.norm` / `map.cmap` etc

Author: oops sorry, didn't notice I wasn't passing `plot_settings` anywhere.
Can't we pass `cmap` as property of `map`?
```python
smap.cmap = plt.cm.Blues_r
fig = plt.figure()
smap.plot(title="My Second Plot")
````
In this way user can adhere to the recent API as well

the same colormap while the data values associated with each color do because
the normalization is different.

> Since, both colormap and the normalization are map-source specific, there is an option to change

Reviewer: ```suggestion
You may also explicitly set both the colormap and normalization::
```

> then from the map itself as follows::

Reviewer: ```suggestion
```
